File: src/main/java/com/sber/yamlprocessor/model/Result.java
package com.sber.yamlprocessor.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public class Result {
    
    @JsonProperty("input-scenarios")
    @NotNull(message = "Input scenarios is required")
    private List<String> inputScenarios;
    
    @Valid
    @NotNull(message = "Reverse is required")
    private List<Reverse> reverse;

    public Result() {}

    public Result(List<String> inputScenarios, List<Reverse> reverse) {
        this.inputScenarios = inputScenarios;
        this.reverse = reverse;
    }

    // Getters and Setters
    public List<String> getInputScenarios() {
        return inputScenarios;
    }

    public void setInputScenarios(List<String> inputScenarios) {
        this.inputScenarios = inputScenarios;
    }

    public List<Reverse> getReverse() {
        return reverse;
    }

    public void setReverse(List<Reverse> reverse) {
        this.reverse = reverse;
    }
}

File: src/main/kotlin/com/sber/yamlprocessor/model/Result.kt
package com.sber.yamlprocessor.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.CascadeType
import jakarta.persistence.CollectionTable
import jakarta.persistence.Column
import jakarta.persistence.ElementCollection
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.OrderColumn
import jakarta.persistence.Table

@Entity
@Table(name = "result")
class Result(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    @Column(name = "id")
    var dbId: Long? = null,

    @ManyToOne(optional = false)
    @JoinColumn(name = "configurator_id")
    @JsonIgnore
    var configurator: Configurator? = null,

    @ElementCollection(fetch = FetchType.LAZY)
    @CollectionTable(name = "result_input_scenario", joinColumns = [JoinColumn(name = "result_id")])
    @Column(name = "scenario", length = 2000)
    @OrderColumn(name = "scenario_sort")
    @JsonProperty("input-scenarios")
    var inputScenarios: MutableList<String> = mutableListOf(),

    @OneToMany(mappedBy = "result", cascade = [CascadeType.ALL], orphanRemoval = true)
    @OrderColumn(name = "reverse_sort")
    var reverse: MutableList<Reverse> = mutableListOf()
)
